Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationAttributeUtil.java

```diff
@@ -59,28 +62,58 @@ private static String getEmailAddressAttrDisplayName() {
         return Bundle.CorrelationAttributeUtil_emailaddresses_text();
     }
 
+    // Defines which artifact types act as the sources for CR data.
+    // Most notably, does not include KEYWORD HIT, CALLLOGS, MESSAGES, CONTACTS
+    // TSK_INTERESTING_ARTIFACT_HIT (See JIRA-6129 for more details on the
+    // interesting artifact hit).
+    private static final Set<Integer> SOURCE_TYPES_FOR_CR_INSERT = new HashSet<Integer>() {{
```

Comment:
Motivation for defining the types, and not their negation (by that I mean not keyword hit, not calllog, not contacts), is this will be safer to keep it up to date. Basically, if a new type comes along for correlation only, we are good. If a new type comes along for insert, during testing they won't see the expected behavior and update this set. However, with the negation, any new artifact types accepted for correlation will automatically be saved, which is too accepting for accidents.

Reply:
I think a comment here to indicate that this list should be updated with any new artifacts that may be source for CR attributes (for creation) would be useful.

Reply:
Good idea, I've added a comment. Thanks.